src/global-messages/GlobalMessagesContext.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import firestore, {
+  FirebaseFirestoreTypes,
+} from '@react-native-firebase/firestore';
+import {LanguageAndText} from '@atb/reference-data/types';
+import {Statuses} from '@atb/theme';
+
+export type GlobalMessage = {
+  id: string;
+  active: boolean;
+  title?: LanguageAndText[];
+  body: LanguageAndText[];
+  type: Statuses;
+  context: GlobalMessageContext[];
+};
+
+export type GlobalMessageContext =
+  | 'app-assistant'
+  | 'app-departures'
+  | 'app-ticketing'
+  | 'web-ticketing'
+  | 'web-overview';
+
+type GlobalMessageContextState = {
+  findGlobalMessages: (context: GlobalMessageContext) => GlobalMessage[];
+};
+
+const defaultGlobalMessageState = {
+  findGlobalMessages: () => [],
+};
+
+const GlobalMessagesContext = createContext<GlobalMessageContextState>(
+  defaultGlobalMessageState,
+);
+
+const GlobalMessagesContextProvider: React.FC = ({children}) => {
+  const [globalMessages, setGlobalMessages] = useState<GlobalMessage[]>([]);
+  const [error, setError] = useState(false);
+
+  useEffect(
+    () =>
+      firestore()
+        .collection('globalMessages')
+        .where('active', '==', true)
+        .onSnapshot(
+          (snapshot) => {
+            const globalMessages = (
+              snapshot as FirebaseFirestoreTypes.QuerySnapshot<GlobalMessage>
+            ).docs.map<GlobalMessage>((d) => d.data());
+            setGlobalMessages(globalMessages);
+            setError(false);
```

Comment:
Maybe for another time, since this is looks just as good as before, but it would be nice to verify the data we get from Firestore more thoroughly than this. Especially since non-developers are going to add messages to Firestore.

Me and @mikaelbr talked about this for the webhop, and I think most of this logic can be copied from here: [client.ts](https://github.com/AtB-AS/webshop2/blob/main/src/global-messages/firebase/client.ts) and [converters.ts](https://github.com/AtB-AS/webshop2/blob/main/src/global-messages/firebase/converters.ts).

Reply:
The verify schema code seems like it can pretty much be copy-pasted, so I think this can be included. Not crashing the app on mistypings is a huge benefit!